skxray/core/roi.py

```diff
@@ -359,137 +357,80 @@ def roi_pixel_values(image, labels, index=None):
-    """
-    Mean intensities for ROIS' of the labeled array for different image sets
+def mean_intensity(images, labeled_array, index=None):
+    """Compute the mean intensity for each ROI in the image list
 
     Parameters
     ----------
-    images_set : array
-        images sets
-        shapes is: (len(images_sets), )
-        one images_set is iterable of 2D arrays dimensions are: (rr, cc)
-
-    labels : array
+    images : list
+        List of images
+    labeled_array : array
         labeled array; 0 is background.
-        Each ROI is represented by a distinct label (i.e., integer).
-
-    Returns
-    -------
-    mean_intensity_list : list
-        average intensity of each ROI as a list
-        shape len(images_sets)
-
-    index_list : list
-        labels list for each image set
-
-    """
-    return tuple(map(list,
-                     zip(*[mean_intensity(im,
-                                          labels) for im in images_set])))
-
-
-def mean_intensity(images, labels, index=None):
-    """
-    Mean intensities for ROIS' of the labeled array for set of images
-
-    Parameters
-    ----------
-    images : array
-        Intensity array of the images
-        dimensions are: (num_img, num_rows, num_cols)
-
-    labels : array
-        labeled array; 0 is background.
-        Each ROI is represented by a distinct label (i.e., integer).
-
-    index : list
-        labels list
-        eg: 5 ROI's
-        index = [1, 2, 3, 4, 5]
+        Each ROI is represented by a nonzero integer. It is not required that
+        the ROI labels are contiguous
+    index : int, list, optional
+        The ROI's to use. If None, this function will extract averages for all
+        ROIs
 
     Returns
     -------
     mean_intensity : array
-        mean intensity of each ROI for the set of images as an array
-        shape (len(images), number of labels)
-
+        The mean intensity of each ROI for all `images`
+        Dimensions:
+            len(mean_intensity) == len(index)
+            len(mean_intensity[0]) == len(images)
+    index : list
+        The labels for each element of the `mean_intensity` list
     """
-    if labels.shape != images[0].shape[0:]:
-        raise ValueError("Shape of the images should be equal to"
-                         " shape of the label array")
+    if labeled_array.shape != images[0].shape[0:]:
+        raise ValueError(
+            "`images` shape (%s) needs to be equal to the labeled_array shape"
+            "(%s)" % (images[0].shape, labeled_array.shape))
+    # handle various input for `index`
     if index is None:
-        index = np.arange(1, np.max(labels) + 1)
-
-    mean_intensity = np.zeros((images.shape[0], index.shape[0]))
+        index = list(np.unique(labeled_array))
+        index.remove(0)
+    try:
+        len(index)
+    except TypeError:
+        index = [index]
+    # not sure that this is needed
+    index = np.asarray(index)
```

Comment:
This line is not needed. The index argument to `ndim.mean` just needs to be a sequence, not a numpy array